src/Request.php

```diff
@@ -132,7 +132,11 @@ public function handleData($data)
         $this->buffer .= $data;
 
         if (false !== strpos($this->buffer, "\r\n\r\n")) {
-            list($response, $bodyChunk) = $this->parseResponse($this->buffer);
+            try {
+                list($response, $bodyChunk) = $this->parseResponse($this->buffer);
+            } catch (\InvalidArgumentException $exception) {
+                $this->emit('error', [$exception, $this]);
+            }
```

Comment:
@djagya It looks like your analysis may be right and makes perfect sense here, sorry for the confusion :+1: Can you update this again?